### PR TITLE
Us11

### LINKED ISSUE #11: Top 5 Merchants By Revenue
As an admin,
When I visit the admin merchants index
Then I see the names of the top 5 merchants by total revenue generated
And I see that each merchant name links to the admin merchant show page for that merchant
And I see the total revenue generated next to each merchant name

Notes on Revenue Calculation:
- Only invoices with at least one successful transaction should count towards revenue
- Revenue for an invoice should be calculated as the sum of the revenue of all invoice items
- Revenue for an invoice item should be calculated as the invoice item unit price multiplied by the quantity (do not use the item unit price)

### PR DESCRIPTION
**Pull request by gregoryjflaherty**
## Description of Changes

closes #11  
* Created AR method top_five_merchants plus model spec in Merchant class
* Created feature tests for view, displaying top_five merchants by revenue 

## Concerns (tags recommended)

* None currently all tests pass 

## What gif describes how you feel about this PR?

jerry_tired

### CODE REVIEW

File: spec/features/merchant_items/index_spec.rb [hunk 1]
```diff
@@ -117,4 +117,87 @@
...
+          expect(@converse.name).to appear_before(@polo.name)
+          expect(@polo.name).to appear_before(@vans.name)
+        end
+      end
+
+      it 'shows revenue' do
+        visit admin_merchants_path
+        expect(current_path).to eq(admin_merchants_path)
+
+        within "div.top_five_merchants" do
```
Comment (jgrazulis) [inline spec/features/merchant_items/index_spec.rb:182]: Nice use of within blocks

**Review by jgrazulis — APPROVED** [review]
Looks great! I think we will have to look at the money conversions for unit prices... not sure if that's already come up.

File: app/models/merchant.rb [hunk 2]
```diff
@@ -7,6 +7,14 @@ class Merchant < ApplicationRecord
 
   validates :name, presence: true
 
+  def self.top_five_by_revenue
+    joins(:items, :invoices, :transactions)
+      .where(transactions: {result: :success})
```
Comment (glsorensen) [inline app/models/merchant.rb:12]: Nice, another meaty one

File: spec/models/merchant_spec.rb [hunk 3]
```diff
@@ -77,4 +85,60 @@
...
+        @transaction_5 = Transaction.create(invoice_id: @invoice_item_5.invoice.id, result: 1)
+        @transaction_6 = Transaction.create(invoice_id: @invoice_item_6.invoice.id, result: 1)
+      end
+
+      it 'gets top 5 merchants by most revenue' do
+        results = Merchant.top_five_by_revenue
+        expect(results[0].name).to eq(@nike.name)
+        expect(results[1].name).to eq(@timberland.name)
+        expect(results[2].name).to eq(@adidas.name)
+        expect(results[3].name).to eq(@converse.name)
```
Comment (glsorensen) [inline spec/models/merchant_spec.rb:139]: robust testing here!

**Review by glsorensen — APPROVED** [review]

**Pull request merged**